WooCommerce/Classes/ViewRelated/Keyboard/KeyboardFrameObserver.swift

```diff
@@ -0,0 +1,56 @@
+import UIKit
+
+/// Observes the keyboard frame and notifies its subscriber.
+final class KeyboardFrameObserver {
```

Comment:
My first instinct would be modelling this as a struct, given it id not going to be passed around, and it mostly does calculations with frames.  I am curious... what made you model it as a class?

Reply:
I usually use `class` when using Objective-C API like `NotificationCenter`'s selector in `addObserver`. I just tried with Struct, and changed to use the [block-based `addObserver` (`addObserver(forName:object:queue:using:)`)](https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver). It seems to work (with a bunch of `mutating` functions): 
https://github.com/woocommerce/woocommerce-ios/pull/1430/commits/0bbfdf20d3d8fdac65bbb3bc4a600d406efb1063